app/models/petition.rb

```diff
@@ -156,6 +155,13 @@ def self.counts_by_state
     counts_by_state
   end
 
+  def publish!
+    self.state = Petition::OPEN_STATE
+    self.open_at = Time.zone.now
+    self.closed_at = AppConfig.petition_duration.months.from_now.end_of_day
```

Comment:
Oh, good point.  I did have to handle it in the tests because the in-memory time was ``23:59:59.99999999`` but the db roundtrip gave us ``23:59:59.99990000`` (for some estimate of decimal places).  Any ideas?